tjmonopix2/analysis/corryvreckan_converter.py

```diff
@@ -0,0 +1,95 @@
+#
+# ------------------------------------------------------------
+# Copyright (c) All rights reserved
+# SiLab, Institute of Physics, University of Bonn
+# ------------------------------------------------------------
+#
+
+from pathlib import Path
+
+import numpy as np
+import tables as tb
+
+from tjmonopix2.analysis import analysis
+
+
+def format_dut(input_filename: str | Path, output_filename: str | Path = None, trigger_mode: str = "AIDA") -> None:
+    """Format hit table to be compatible with corryvreckan EventLoaderHDF5 as of commit 149e937f
+
+    Parameters
+    ----------
+    input_filename : str | Path
+        Interpreted hit file with Hit and/or Event data depending on trigger_mode
+    output_filename : str | Path, optional
+         Output file name. If None, defaults to input_filename with suffix '_converted', by default None
+    trigger_mode : str, optional
+        Trigger mode during data taking. EUDET mode (with trigger handshake) and AIDA, mode (without
+        handshake only) are supported. In general, use `DATA_FORMAT=1` in `testbench.yaml` for data
+        taking. By default "AIDA"
+
+    Raises
+    ------
+    RuntimeError
+        Invalid trigger mode selected
+    """
+    hit_dtype_converted = np.dtype(
+        [("column", "<u2"), ("row", "<u2"), ("charge", "<u2"), ("timestamp", "<u8"), ("trigger_number", "<u4")]
+    )
+
+    if type(input_filename) is str:
+        input_filename = Path(input_filename)
+    if output_filename is None:
+        output_filename = Path(input_filename.parent) / Path(input_filename.stem + "_converted" + input_filename.suffix)
+    with tb.open_file(input_filename, "r") as in_file:
+        if trigger_mode.lower() == "aida":
+            hit_table_in = in_file.root.Dut[:]
+            sel = hit_table_in["col"] <= 512  # Select only DUT words
+            hits_selected = hit_table_in[sel]
+
+            hit_table_converted = np.zeros(len(hits_selected), dtype=hit_dtype_converted)
+            with tb.open_file(output_filename, "w") as out_file:
+                hit_table_out = out_file.create_table(out_file.root, name="Hits", description=hit_dtype_converted)
+                # Internal column and numbers start at 0 TODO: check with corry
+                hit_table_converted["column"] = hits_selected["col"] + 1
```

Comment:
Is this really the case that corry starts counting at 1? I didn't do that in my tests, and it somewhat looks strange to me...

Reply:
I have no idea, actually. I will look it up and remove the TODO notice.

Reply:
I had a look into our loader of the eudaq .raw files, and there we count from zero as usual (if i looked into the code correctly)

Reply:
Thanks for checking, I will remove it and keep the first column and row number as 0. After all, it is a dedicated corryvreckan converter and must only be compatible with that framework.